lib/services/resource.go

```diff
@@ -111,6 +115,13 @@ func PreserveResourceID() MarshalOption {
 	}
 }
 
+func SkipCheckAndSetDefaults() MarshalOption {
```

Comment:
I don't really understand the point of the functional options pattern here if the underlying field is already exported, but this looks to be a common pattern in this file.

Reply:
I've removed this function, so I'll resolve.